### PR TITLE
Kotlin migration 2

### PR DESCRIPTION
**Pull request by kormateusz**

### CODE REVIEW

File: library/src/main/java/com/applandeo/materialcalendarview/adapters/CalendarDayAdapter.kt [hunk 1]
```diff
@@ -17,100 +18,100 @@ import java.util.*
...
+
         // Setting current month day color
         day.setCurrentMonthDayColors(today, dayLabel, calendarProperties)
     }
 
-    private fun isSelectedDay(day: Calendar) =
-            (calendarProperties.calendarType != CalendarView.CLASSIC
-                    && day.get(Calendar.MONTH) == pageMonth
-                    && calendarPageAdapter.selectedDays.contains(SelectedDay(day)))
+    private fun Calendar.isSelectedDay(): Boolean {
```
Comment (Afur) [inline library/src/main/java/com/applandeo/materialcalendarview/adapters/CalendarDayAdapter.kt]:
```
private fun Calendar.isSelectedDay(): Boolean =
        calendarProperties.calendarType != CalendarView.CLASSIC
                && this[Calendar.MONTH] == pageMonth
                && SelectedDay(this) in calendarPageAdapter.selectedDays
```

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt [hunk 2]
```diff
@@ -105,43 +110,60 @@ class DayRowClickListener(
...
-        selectedDay.calendar?.setCurrentMonthDayColors(midnightCalendar, selectedDay.view, calendarProperties)
+        selectedDay.calendar.setCurrentMonthDayColors(
+                midnightCalendar,
+                selectedDay.view as? TextView,
+                calendarProperties
+        )
     }
 
-    private fun isCurrentMonthDay(day: Calendar) = day.get(Calendar.MONTH) == pageMonth && isBetweenMinAndMax(day)
+    private fun Calendar.isCurrentMonthDay(): Boolean {
```
Comment (Afur) [inline library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]:
```
    private fun Calendar.isCurrentMonthDay(): Boolean =
        this[Calendar.MONTH] == pageMonth && this.isBetweenMinAndMax()
```

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+        if (isAnotherDaySelected(previousSelectedDay, day)) {
+            selectDay(dayLabel, day)
+            reverseUnselectedColor(previousSelectedDay)
+        }
+    }
+
+    private fun selectManyDays(view: View, day: Calendar) {
+        val dayLabel = view.findViewById<TextView>(R.id.dayLabel)
+
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]:
```
if (!day.isCurrentMonthDay() || !day.isActiveDay()) {
return
}
```

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            }
+
+            calendarPageAdapter.addSelectedDay(selectedDay)
+        }
+    }
+
+    private fun selectRange(view: View, day: Calendar) {
+        val dayLabel = view.findViewById<TextView>(R.id.dayLabel)
+
+        if (!day.isCurrentMonthDay() || !day.isActiveDay()) return
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]: `        if (!day.isCurrentMonthDay() || !day.isActiveDay()) {return}
`

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    private fun selectOneAndRange(dayLabel: TextView, day: Calendar) {
+        val previousSelectedDayCalendar = calendarPageAdapter.selectedDay.calendar
+
+        previousSelectedDayCalendar.getDatesRange(day)
+                .filter { it !in calendarProperties.disabledDays }
+                .forEach { calendarPageAdapter.addSelectedDay(SelectedDay(it)) }
+
+        if (isOutOfMaxRange(previousSelectedDayCalendar, day)) return
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]: `        if (isOutOfMaxRange(previousSelectedDayCalendar, day)){return}
`

File: library/src/main/java/com/applandeo/materialcalendarview/utils/AppearanceUtils.kt [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    labels.forEachIndexed { index, label ->
+        label.text = abbreviations[(index + firstDayOfWeek - 1) % 7]
+
+        if (color != 0) {
+            label.setTextColor(color)
+        }
+    }
+}
+
+fun View.setHeaderColor(color: Int) {
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/utils/AppearanceUtils.kt]: What do you think will be better :

```
 if (color == 0) return
    this.calendarHeader.setBackgroundColor(color)
```

OR

```
 if (color =!  0) {
    this.calendarHeader.setBackgroundColor(color)
}
```

File: library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * This helper class represent a selected day when calendar is in a picker date mode.
+ * It is used to remember a selected calendar cell.
+ *
+ * @param view     View representing selected calendar cell
+ * @param calendar Calendar instance representing selected cell date
+ *
+ * @return View representing selected calendar cell
+ *
+ * Created by Applandeo Team.
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt]: ?

File: library/src/main/java/com/applandeo/materialcalendarview/builders/DatePickerBuilder.kt [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+import com.applandeo.materialcalendarview.listeners.OnCalendarPageChangeListener
+import com.applandeo.materialcalendarview.listeners.OnSelectDateListener
+import com.applandeo.materialcalendarview.utils.CalendarProperties
+import java.util.*
+
+
+/**
+ * Created by Applandeo Team.
+ */
+class DatePickerBuilder(
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/builders/DatePickerBuilder.kt]: Builder: https://www.baeldung.com/kotlin-builder-pattern

File: library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+
+    fun setOnPreviousPageChangeListener(listener: OnCalendarPageChangeListener) {
+        calendarProperties.onPreviousPageChangeListener = listener
+    }
+
+    fun setOnForwardPageChangeListener(listener: OnCalendarPageChangeListener) {
+        calendarProperties.onForwardPageChangeListener = listener
+    }
+
+    private fun isScrollingLimited(calendar: Calendar, position: Int): Boolean {
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt]: if else if statement?

File: library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+    fun setOnDayClickListener(onDayClickListener: OnDayClickListener) {
+        calendarProperties.onDayClickListener = onDayClickListener
+    }
+
+    /**
+     * This method set a current and selected date of the calendar using Calendar object.
+     *
+     * @param date A Calendar object representing a date to which the calendar will be set
+     */
+    @Throws(OutOfDateRangeException::class)
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt]: Add in header that it's throw exception

File: library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+     */
+    @Deprecated("Use getFirstSelectedDate()", ReplaceWith("firstSelectedDate"))
+    val selectedDate: Calendar
+        get() = firstSelectedDate
+
+    /**
+     * @return Calendar object representing a selected date
+     */
+    val firstSelectedDate: Calendar
+        get() {
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt]: get () = ?

File: library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt [hunk 12]
```diff
@@ -0,0 +1,399 @@
...
+            return calendarPageAdapter.selectedDays.map { it.calendar }.first()
+        }
+
+    /**
+     * @return Calendar object representing a date of current calendar page
+     */
+    val currentPageDate: Calendar
+        get() {
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/CalendarView.kt]: get() = ?

File: library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt (hunk 7 above)
Comment (kormateusz) [reply library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt]: I do not understand my friend 🤔

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+        if (isAnotherDaySelected(previousSelectedDay, day)) {
+            selectDay(dayLabel, day)
+            reverseUnselectedColor(previousSelectedDay)
+        }
+    }
+
+    private fun selectManyDays(view: View, day: Calendar) {
+        val dayLabel = view.findViewById<TextView>(R.id.dayLabel)
+
+        if (day.isCurrentMonthDay() && day.isActiveDay()) {
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]: you can do it in the same way like below:

`if (!day.isCurrentMonthDay() || !day.isActiveDay()) return`

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt [hunk 14]
```diff
@@ -0,0 +1,162 @@
...
+        }
+
+        val eventDay = calendarProperties.eventDays.firstOrNull { it.calendar == day }
+
+        if (eventDay == null) {
+            callOnClickListener(EventDay(day))
+            return
+        }
+
+        callOnClickListener(eventDay)
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]: `        callOnClickListener(eventDay ?:  EventDay(day))
`

File: library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.*
+
+/**
+ * This helper class represent a selected day when calendar is in a picker date mode.
+ * It is used to remember a selected calendar cell.
+ *
+ * @param calendar Calendar instance representing selected cell date
+ * @param view     View representing selected calendar cell
+ *
+ * Created by Applandeo Team.
```
Comment (mateuszSawa) [inline library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt]: Remove: `Created by Applandeo Team.`

File: library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt (hunk 14 above)
Comment (mateuszSawa) [reply library/src/main/java/com/applandeo/materialcalendarview/listeners/DayRowClickListener.kt]: or  

```
val eventDay = calendarProperties.eventDays.firstOrNull { it.calendar == day } ?: EventDay(day)
callOnClickListener(eventDay)
```

File: library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt (hunk 15 above)
Comment (kormateusz) [reply library/src/main/java/com/applandeo/materialcalendarview/utils/SelectedDay.kt]: Everywhere or just here?

**Pull request merged**